sdk/daml-script/test/daml/upgrades/Interfaces.daml

```diff
@@ -163,16 +183,29 @@ main = tests
   , ("Multiple interface exercises use their respective package preference entries at choice body level", multipleInterfaceChoicePackagePreferenceUpdate)
   , ("queryInterfaceContractId works as intended in the presence of upgraded packages", queryInterfaceUpgrades)
   , ("fetchFromInterface works as intended in the presence of upgraded packages", fetchFromInterfaceUpgrades)
+  , ("Calling a new interface choice uses the highest version implementation at command level by default", defaultHighestVersionChoiceImplementationForNewInterfaceCommand)
+  , ("Calling a new interface choice uses the highest version implementation at choice body level by default", defaultHighestVersionChoiceImplementationForNewInterfaceUpdate)
+  , ("Package map preference is used for new interface implementation selection at command level", newInterfaceChoicePackagePreferenceCommand)
+  , ("Package map preference is used for new interface implementation selection at choice body level", newInterfaceChoicePackagePreferenceUpdate)
+  , broken ("queryInterfaceContractId works as intended in the presence of upgraded packages with new interface instances", queryNewInterfaceUpgrades)
```

Comment:
Do we test somewhere that the highest or preferred version of an interface is used?

I suspect we need to ensure that we upgrade to the target templateId on a fetch.

Reply:
:+1: I'll put together a test for that now, but it may fall under the same issues as queryInterface, in which case I'll push that work to a next PR and mark the test as broken.

Reply:
Interface fetch works, downgrading/upgrading v3 to/from v2 works, marking this as done.